website_mrp_bom/controllers/main.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2018 Eficent Business and IT Consulting Services S.L.
+#   (http://www.eficent.com)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+from odoo import http
+from odoo.http import request
+
+
+class WebsiteMrpBomPage(http.Controller):
+
+    @http.route(['/components/<model("mrp.bom"):bom>'], type='http',
+                auth="public", website=True)
+    def boms_detail(self, bom=False, **post):
+        if bom and (
+                bom.website_published or self._is_website_publisher()):
+            values = {
+                'main_object': bom,
+                'bom': bom,
+                'has_published_bom': self._has_published_bom(bom)
+            }
+            return request.render("website_mrp_bom.mrp_bom_page", values)
+        return request.not_found()
+
+    def _is_website_publisher(self):
+        return self.env['res.users'].has_group(
```

Comment:
shouldn't this be `request.env.user.has_group( 'website.group_website_publisher')`